### DISCUSSION TITLE
Pihole on BSD

### DISCUSSION THREAD

**Discussion by mihakralj** [mimugmail/opn-repo#47] [Ideas]
There is an (unfinished) experiment to port FTL to BSD - the only component of pihole that doesn't exist in BSD domain. 

https://github.com/freqlabs/FTL (there is freebsd-wip branch)

https://discourse.pi-hole.net/t/freebsd-compatability/2092/46

Worth picking it up and seeing if we can put the humpty dumpy back together on FreeBSD.

**Comment by mimugmail** [comment]
Hm, currently I dont have the time to play around with it, but I'm the last one stopping it for integration here :)

**Reply by mihakralj** [reply]
I complied the FTL binary successfully tonight and can now run DNS queries against it on my OPNSenese. Now it needs all suppoting systems and configs and .sh files that compose the full pihole.

I actually wonder if I should integrate the pihole UI into OPNsense Phalcon... 😊

**Comment by mihakralj** [comment]
Well, I have some good news and a question how to proceed.

Good news: **the dang thing works on OPNSense!** Well, it kinda works. I mean, all individual components work. The compiled binary **pihole-FTL** runs as a service with not much of a complaint - after it is padded with all required databases, config files and whatnot. I started to clean the pihole.sh to use BSD instead of Linux commands. See screenshots below.

But there is tons (and tons and tons) of work to do:
- all .sh scripts need to be combed through and de-linuxized (dir structure, commands that don't exist in BSD)
- we need to rip out things we don't want Pihole to do on OPNsense - like dhcp and lighttpd
- PHP pages for Pihole can easily co-exist on OPNsense in `/usr/local/www/pihole `and `/usr/local/www/admin` - but need lots of cleanup to de-linuxize all system calls in them

So, this is not a small job, but very doable - since all components exist, work and just need careful cleanup and assembly. Should be a larger group effort - or else it will drown one or two lone enthusiasts.

Worth it?

**Comment by mihakralj** [comment]
![image](https://user-images.githubusercontent.com/31756078/117390994-fa77f580-aea3-11eb-92c0-0e6ad875986c.png)
![image](https://user-images.githubusercontent.com/31756078/117391038-0cf22f00-aea4-11eb-82a4-fc30acac0545.png)
![image](https://user-images.githubusercontent.com/31756078/117391531-f5677600-aea4-11eb-93f1-51acecdfd680.png)

**Comment by gspannu** [comment]
Any update on this project? 

With the new version of PiHole (without lighttpd), any chance of seeing this being available in OPNsense?

A lot of people use AGH, but the PiHole reporting is still unmatched...

Thanks...
